bitnami/grafana-tempo/templates/scaling-monolithic/headless-service.yaml

```diff
@@ -0,0 +1,97 @@
+{{- /*
+Copyright Broadcom, Inc. All Rights Reserved.
+SPDX-License-Identifier: APACHE-2.0
+*/}}
+
+{{- if eq .Values.tempo.deploymentMode "scalingMonolithic" }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "grafana-tempo.scaling-monolithic.fullname" . }}-headless
+  namespace: {{ .Release.Namespace | quote }}
+  {{- $versionLabel := dict "app.kubernetes.io/version" ( include "common.images.version" ( dict "imageRoot" .Values.scalingMonolithic.query.image "chart" .Chart ) ) }}
+  {{- $labels := include "common.tplvalues.merge" ( dict "values" ( list .Values.commonLabels $versionLabel ) "context" . ) }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" $labels "context" $ ) | nindent 4 }}
+    app.kubernetes.io/component: scaling-monolithic
+  {{- if or .Values.commonAnnotations .Values.scalingMonolithic.service.headless.annotations }}
+  {{- $annotations := include "common.tplvalues.merge" ( dict "values" ( list .Values.scalingMonolithic.service.headless.annotations .Values.commonAnnotations ) "context" . ) }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" $annotations "context" $) | nindent 4 }}
+  {{- end }}
+spec:
+  # NOTE: this is based on this https://github.com/grafana/helm-charts/pull/451. Should be reviewed if it is actually necessary in the future
+  publishNotReadyAddresses: true
+  type: ClusterIP
+  clusterIP: None
+  ports:
+    - name: http
+      port: {{ .Values.scalingMonolithic.service.ports.http }}
+      targetPort: http
+      protocol: TCP
+    - name: grpc
+      port: {{ .Values.scalingMonolithic.service.ports.grpc }}
+      targetPort: grpc
+      protocol: TCP
+    ## From here we use non-configurable, standard ports
```

Comment:
I'm not sure about having all these ports here. As far as I understand, the headless service should only have the Query Frontend's service ports as we currently have in the `templates/query-frontend/headless-service.yaml` file. Are the rest of ports needed?

Reply:
That was to keep align with the ClusterIP service, so `extraPorts` can affect both  the ClusterIP service and the headless service.
I changed this, to keep minimal required ports. And added new variable in `values.yaml`, to allow to set extra ports for headless service specifically